runtime/lua/vim/iter.lua

```diff
@@ -1052,6 +1120,23 @@ function M.map(f, src, ...)
   return Iter.new(src, ...):map(f):totable()
 end
 
+--- Flattens a |list-iterator|, un-nesting nested values
+---
+--- ```lua
+--- -- Equivalent to:
+--- vim.iter(src):flat():totable()
+--- ```
+---
+---@see |Iter:flat()|
+---
+---@param src table|function Table or iterator function to flatten
```

Comment:
Only list-like tables are accepted:

```suggestion
---@param src table List-like table to flatten
```